parser/src/aspects/call.rs

```diff
@@ -46,45 +46,64 @@ pub trait CallAspect<'a> {
 
 impl<'a> CallAspect<'a> for Parser<'a> {
     fn any_call(&mut self) -> ParseResult<Expr<'a>> {
-        let path = self.parse_inclusion_path()?;
+        let mut path = self.parse_inclusion_path()?;
 
         // Equivalent to #may_be_at_programmatic_call_start, with an additional check for lambda definitions.
         if self
             .cursor
             .lookahead(
-                of_type(Identifier).and_then(
-                    of_types(&[TokenType::RoundedLeftBracket, TokenType::SquaredLeftBracket])
-                        .or(spaces().then(of_type(TokenType::FatArrow))),
+                any().and_then(
+                    of_types(&[
+                        TokenType::RoundedLeftBracket,
+                        TokenType::SquaredLeftBracket,
+                        TokenType::ColonColon,
+                    ])
+                    .or(spaces().then(of_type(TokenType::FatArrow))),
                 ),
             )
             .is_none()
         {
-            ensure_empty("Command calls cannot have inclusion path", path, |str| {
-                self.cursor.relative_pos(*str)
-            })?;
+            ensure_empty(
+                "Command calls cannot have inclusion path",
+                path,
+                InclusionPathItem::segment,
+            )?;
             return self.call();
         }
         // We don't known if this is a programmatic call, a raw call or a lambda definition yet.
 
         let identifier = self.cursor.next()?;
-        let value = identifier.value;
+        let name = identifier.value;
+        let name_segment = self.cursor.relative_pos(name);
         let callee = Expr::Literal(Literal {
-            parsed: value.into(),
-            segment: self.cursor.relative_pos(value),
+            parsed: name.into(),
+            segment: name_segment.clone(),
         });
         let type_parameters = self.parse_type_parameter_list()?.0;
         if let Some(open_parenthesis) = self.cursor.advance(of_type(TokenType::RoundedLeftBracket))
         {
+            if identifier.token_type != TokenType::Identifier {
+                return self.expected_with(
+                    "Expected function name.",
+                    identifier,
+                    ParseErrorKind::Expected("valid function identifier".to_string()),
```

Comment:
```suggestion
                    ParseErrorKind::Expected("identifier".to_owned()),
```
All identifiers are valid.